lib/oli/activities/reports/providers/OliLikert.ex

```diff
@@ -0,0 +1,172 @@
+defmodule Oli.Activities.Reports.Providers.OliLikert do
+  @behaviour Oli.Activities.Reports.Renderer
+
+  alias Oli.Delivery.Attempts.Core
+
+  @impl Oli.Activities.Reports.Renderer
+  def render(
+        %Oli.Rendering.Context{enrollment: enrollment, user: user} = context,
+        %{"activityId" => activity_id} = _element
+      ) do
+    with %{choices: choices, data: data} <-
+           process_attempt_data(enrollment.section_id, user.id, activity_id) do
+      data_url = "/api/v1/activity/report/#{enrollment.section_id}/#{activity_id}"
```

Comment:
We shouldn't hardcode URLs in code.  Instead, use the `Routes` module to determine the url.  But doing so here would lead to a dependency from `Oli` to `OliWeb`, which we don't want.   So you will need to pass to this function a function that can build the URL.

Reply:
The purpose of splitting the data apart from the visualization and instead embedding a URL to the controller was so that we do not have to do any DB calls during the rendering phase.  But `process_attempt_data` ends up making three DB queries.  One for the "get_latest_attempt", then two for the external `preloads`.  

We need to remove all of these DB calls here.

Reply:
No hardcoded URL in the backend. Moved logic into a new likert frontend component

I re-implemented it with a new solution that avoids most db calls except for a minor one on the render function.